data/helpers.d/php

```diff
@@ -2,18 +2,47 @@
 
 # Create a dedicated php-fpm config
 #
-# usage: ynh_add_fpm_config [--phpversion=7.X]
+# usage 1: ynh_add_fpm_config [--phpversion=7.X] [--use_template]
 # | arg: -v, --phpversion - Version of php to use.
+# | arg: -t, --use_template - Use this helper in template mode.
+#
+# -----------------------------------------------------------------------------
+#
+# usage 2: ynh_add_fpm_config [--phpversion=7.X] --usage=usage --footprint=footprint
+# | arg: -v, --phpversion - Version of php to use.#
+# | arg: -f, --footprint      - Memory footprint of the service (low/medium/high).
+# low    - Less than 20Mb of ram by pool.
+# medium - Between 20Mb and 40Mb of ram by pool.
+# high   - More than 40Mb of ram by pool.
+# Or specify exactly the footprint, the load of the service as Mb by pool instead of having a standard value.
+# To have this value, use the following command and stress the service.
+# watch -n0.5 ps -o user,cmd,%cpu,rss -u APP
+#
+# | arg: -u, --usage     - Expected usage of the service (low/medium/high).
+# low    - Personal usage, behind the sso.
+# medium - Low usage, few people or/and publicly accessible.
+# high   - High usage, frequently visited website.
 #
 # Requires YunoHost version 2.7.2 or higher.
 ynh_add_fpm_config () {
 	# Declare an array to define the options of this helper.
-	local legacy_args=v
-	declare -Ar args_array=( [v]=phpversion= )
+	local legacy_args=vtuf
+	declare -Ar args_array=( [v]=phpversion= [t]=use_template [u]=usage= [f]=footprint= )
 	local phpversion
+	local use_template
+	local usage
+	local footprint
 	# Manage arguments with getopts
 	ynh_handle_getopts_args "$@"
 
+	# The default behaviour is to use the template.
+	use_template="${use_template:-1}"
+	usage="${usage:-}"
+	footprint="${footprint:-}"
+	if [ -n "$usage" ] || [ -n "$footprint" ]; then
+		use_template=0
+	fi
```

Comment:
Hmokay, but then is it really so useful to define this use_template argument ? Can't we just have it as an internal variable that is false only if `usage` or `footprint` is defined ..?

Reply:
Same concept than that previous PR https://github.com/YunoHost/yunohost/pull/364/commits/bba393c45c5ee45a942f6c7be697e7a2415e14b1 on fail2ban.

Which was actually in the other way, a template option was added.
We indeed don't **need** that variable, but it think it's nice to have it to make the code clear.

The argument on its own is not mandatory, its default value is 1.